src/ADAL.PCL.Desktop/WindowsFormsWebAuthenticationDialogBase.cs

```diff
@@ -217,19 +228,19 @@ private bool CheckForClosingUrl(Uri url)
         {
             // Make change here
             bool canClose = false;
-            if (url.Authority.Equals(this.desiredCallbackUri.Authority, StringComparison.CurrentCultureIgnoreCase) &&
-                url.AbsolutePath.Equals(this.desiredCallbackUri.AbsolutePath, StringComparison.CurrentCulture))
+            if (url.Authority.Equals(this.desiredCallbackUri.Authority, StringComparison.OrdinalIgnoreCase) &&
+                url.AbsolutePath.Equals(this.desiredCallbackUri.AbsolutePath, StringComparison.OrdinalIgnoreCase))
             {
                 this.Result = new AuthorizationResult(AuthorizationStatus.Success, url.OriginalString);
                 canClose = true;
             }
 
             // if redirect_uri is not hit and scheme of the url is not whitelisted and url is not about:blank
             // and url scheme is not https then fail to load.
-            if (!canClose && !whiteListedSchemes.Contains(url.Scheme.ToLower(CultureInfo.CurrentCulture)) &&
-                !url.AbsoluteUri.Equals("about:blank", StringComparison.CurrentCultureIgnoreCase)
-                && !url.Scheme.Equals("https", StringComparison.CurrentCultureIgnoreCase) 
-                && !url.Scheme.Equals("javascript", StringComparison.CurrentCultureIgnoreCase))
+            if (!canClose && !WhiteListedSchemes.Contains(url.Scheme)
+                && !url.AbsoluteUri.Equals(AboutBlankUri, StringComparison.OrdinalIgnoreCase)
+                && !url.Scheme.Equals(Uri.UriSchemeHttps, StringComparison.OrdinalIgnoreCase)
+                && !url.Scheme.Equals(JavaScriptScheme, StringComparison.OrdinalIgnoreCase))
```

Comment:
Not something you've changed, but it seems odd that there are explicit checks for the _https_ and _javascript_ schemes rather than just whitelisting them.